### ISSUE TITLE
full CTS support

### ISSUE THREAD

**Issue by ManInMyVan** [ViaVersion/ViaFabricPlus#181]
- [ ] generic.attack_reach attribute
- [ ] Cleaving enchantment
- [ ] Shield Indicator
- [ ] Activate shields when sneaking
- [x] 4 tick cooldown on eggs, snowballs
- [ ] Proper attack speeds for weapons
- [ ] Removed warm-up delay from shields
- [ ] Correct damage values
- [ ] Snowballs stack to 64
- [ ] Potions stack to 16.
- [ ] Correct strength potion lore
- [x] Eating and drinking animations are interrupted and reset when being hit by a player/mob
- [ ] The attack cooldown is now only 4 ticks (200ms) when an attack is missed.
- [ ] Attacking is no longer possible until the weapon is recharged.
- [ ] The attack timer is now unaffected by switching items, only resetting when the player performs an attack.
- [ ] Entities with bounding boxes smaller than 0.9 blocks (such as rabbits and bats, for example) now have them inflated to that value.
- [ ] Entities can now be hit through transparent blocks like grass or vines, without destroying said block.
- [ ] The attack key can be held down to automatically attack at the next available moment. (Has a +1 tick (50ms) penalty, making it less effective than optimal timing or clicking.)
- [ ] Weapons with shorter cooldowns give victims a shorter attack invulnerability time (invincibility frames).
- [ ] anything else I missed

**Comment by FlorianMichael** [comment]
You can't join CTS servers in the first place, how did you find out that some of the bullet points are already "done"?

**Comment by ManInMyVan** [comment]
Go to "supported versions" on the readme.md on this repo, and find "snapshots"

**Comment by ManInMyVan** [comment]
> You can't join CTS servers in the first place

Go to "supported versions" on the readme.md on this repo, and find "snapshots"

**Comment by FlorianMichael** [comment]
Ah you mean Combat Test 8C, I thought you meant the Atlas Fork because it is abbreviated CTS

**Comment by CuteistCat** [comment]
why wontfix?

**Comment by FlorianMichael** [comment]
Because no one in the Via team has either the time or motivation to do this, there is barely anyone even playing these april fool versions and it's just not worth spending hours and days implementating these changes, I won't close this issue for now because it's still valid, but unlikely that I will ever do it.

**Comment by Gaming32** [comment]
Combat Tests aren't April Fools versions. They're also have decently large communities around them.

**Comment by FlorianMichael** [comment]
Yes, but I treat them the same way because the protocol for it is part of ViaAprilFools which is already dead because no one cares enough to update it properly.

**Comment by FlorianMichael** [comment]
We might actually be able to implement some of those features now that 1.20.6 provided new attributes.

**Comment by lowercasebtw** [comment]
why hasnt there been more work on this, 1.20.5 added item components, I see it being easy to make snowballs stack to 64 and potions stack to 16, etc

**Comment by lowercasebtw** [comment]
generic.attack_reach isnt that entity_interaction_range from 1.20?

**Comment by florianreuth** [comment]
Most likely not something we'll be working on the future due to limited time available and more important issues.
